### PR TITLE
GH-39779: [Python] Expose force_virtual_addressing in PyArrow

### LINKED ISSUE #39779: [Python] Expose force_virtual_addressing in PyArrow
### Describe the enhancement requested

In #36827 we added the `force_virtual_addressing` option to the S3 implementation in Arrow C++. It should now be exposed to Python users.

### Component(s)

Python

### PR DESCRIPTION
**Pull request by yysd5**
### Rationale for this change / What changes are included in this PR?

To expose force_virtual_addressing in PyArrow.

### Are these changes tested?

Existing unit tests are not broken, and a new test case have been added.

### Are there any user-facing changes?

pyarrow.fs.S3FileSystem: it becomes possible to specify the argument 'force_virtual_addressing'.

* Closes: #39779

### CODE REVIEW

**Comment by AlenkaF** [comment]
Thank you for you contribution @yo1956 !

From a quick look I think that the Cython layer will have to be updated to include the `force_virtual_addressing` 
https://github.com/apache/arrow/blob/21ffd82c05c93b873ae3c27128eb8604ed0c735f/python/pyarrow/includes/libarrow_fs.pxd#L161-L185

**Comment by yo1956** [comment]
@AlenkaF 
Thank you for reviewing!
I'll fix it.

**Comment by yo1956** [comment]
@AlenkaF 
I've made the modifications, but it seems there are errors with CI that doesn't appear to be related to my changes.
I'm not entirely sure, though. I would be grateful if you could provide some advice when you have time.

**Comment by AlenkaF** [comment]
The failures are not related, there is a fix already open: https://github.com/apache/arrow/pull/39827. Similar for the Sphinx build.

We can wait for them to get merged and rebase then.

**Review by AlenkaF — APPROVED** [review]
LGTM, thank you for the updates!

I do not think there are any additions needed in the tests, but would like to have one extra approve before merging.

**Comment by yo1956** [comment]
> LGTM, thank you for the updates!
> 
> I do not think there are any additions needed in the tests, but would like to have one extra approve before merging.

@AlenkaF 
Thank you!!

**Comment by yo1956** [comment]
@jorisvandenbossche 
Could you please review this when you have time?

**Review by jorisvandenbossche — APPROVED** [review]
Looks good, thanks!

**Pull request merged**
